src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final Map<Class<? extends Machine>, MachineProducer<? extends Machine>> producerMap;
+
+    {
+        producerMap = new HashMap<>();
+        producerMap.put(Bulldozer.class, new BulldozerProducer());
+        producerMap.put(Excavator.class, new ExcavatorProducer());
+        producerMap.put(Truck.class, new TruckProducer());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <V extends Machine> List<V> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> producer = producerMap.getOrDefault(type, null);
```

Comment:
Null - is already default 
```suggestion
        MachineProducer<? extends Machine> producer = producerMap.get(type);
```